src/components/input/input.js

```diff
@@ -663,23 +677,11 @@ function mdMaxlengthDirective($animate, $mdUtil) {
           $animate.leave(charCountEl);
         }
       });
-
-      ngModelCtrl.$validators['md-maxlength'] = function(modelValue, viewValue) {
-        if (!angular.isNumber(maxlength) || maxlength < 0) {
-          return true;
-        }
-
-        // We always update the char count, when the modelValue has changed.
-        // Using the $validators for triggering the update works very well.
-        renderCharCount();
-
-        return ( modelValue || element.val() || viewValue || '' ).length <= maxlength;
-      };
     });
 
     function renderCharCount(value) {
       // If we have not been appended to the body yet; do not render
-      if (!charCountEl.parent) {
+      if (!charCountEl || !charCountEl.parent) {
```

Comment:
Is `||` correct here? Should this be `&&`?

Reply:
We now have to check that charCountEl is not undefined or charCountEl.text will fail in the following lines in some cases (because charCountEl is initialized at nextTick and renderCharCount can now be called before nextTick, from validator).

Reply:
OK, can you please update the comment as well?